File: P2/Ejercicio1_MantenimientoFlutter/lib/core/FilterManager.dart
import 'FilterChain.dart';
import '../interfaces/Filter.dart';
import 'AuthenticationService.dart';

class FilterManager {
  late FilterChain _filterChain;

  FilterManager(AuthenticationService target) {
    _filterChain = FilterChain();
    _filterChain.setTarget(target);
  }

  void addFilter(Filter filter) {
    _filterChain.addFilter(filter);
  }

  // procesa la petición y devuelve el resultado de la validación
  String? filterRequest(Map<String, String> request) {
    return _filterChain.execute(request);
  }
}

File: P1/Ejercicio4/core/filter_manager.rb
require_relative 'filter_chain'

class FilterManager
  def initialize(target)
    @filter_chain = FilterChain.new
    @filter_chain.set_target(target)
  end

  # añade un filtro
  def add_filter(filter)
    @filter_chain.add_filter(filter)
  end

  # ejecuta ese filtro
  def filter_request(request)
    @filter_chain.execute(request)
  end
end